File: problem_one/1.rb
def do_magic(max)
  [*3...max].select do |number|
    divisible_by?(number, 3, 5)
  end.inject(:+)
end

def divisible_by?(number, *divisors)
  divisors.each do |divisor|
    return true if number % divisor == 0
  end
  return false
end

p do_magic(1000)

File: problem_one/1.js
function divisibleBy (number, divisors) {
  for (var i=0; i<divisors.length; i++){
    if (number % divisors[i] === 0) {
      return true;
    }
  }
  return false;
}

(function doMagic (max) {
  var results =[];
  for (var i=3; i<=max; i++){
    if (divisibleBy(i, [3,5])) {
      results.push(i);
    }
  }
  var sum = results.reduce(function(a,b){
    return a + b;
  });
  return console.log(sum);
})(1000);